src/lokijs.js

```diff
@@ -3328,14 +3348,7 @@
         throw new Error('Do not know what you want to do.');
       }
 
-      // for regex ops, precompile
-      if (operator === '$regex') {
-        if (Array.isArray(value)) {
-          value = new RegExp(value[0], value[1]);
-        } else if (!(value instanceof RegExp)) {
-          value = new RegExp(value);
-        }
-      }
+      value = precompileQuery(operator, value);
```

Comment:
again, if we could constrain this to only execute when :
```javascript
if (operator === '$regex' || typeof value === 'object') {
```

Reply:
feel free to discuss any of the above comments, they do however improve results of our single object lookups which have primitive values as well as $eq (non $regex) ops.